pkg/cosmos/client.go

```diff
@@ -0,0 +1,95 @@
+package cosmos
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/url"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+// Client is a client for Cosmos.
+type Client struct {
+	http *httpclient.Client
+}
+
+// NewClient creates a new Cosmos client.
+func NewClient(baseClient *httpclient.Client) *Client {
+	return &Client{
+		http: baseClient,
+	}
+}
+
+// PackageInfo contains information about a DC/OS package.
+// It is mirroring the JSON response of the `/package/describe`
+// endpoint for unmarshalling convenience.
+type PackageInfo struct {
+	Package struct {
+		Name        string `json:"name"`
+		Description string `json:"description"`
+		Version     string `json:"version"`
+		Resource    struct {
+			CLI struct {
+				Plugins map[string]map[string]*Plugin `json:"binaries"`
+			} `json:"cli"`
+		} `json:"resource"`
+	} `json:"package"`
+}
+
+// Plugin represents a CLI plugin resource.
+type Plugin struct {
+	Kind string `json:"kind"`
```

Comment:
`Kind` makes sense as it is used in https://docs.mesosphere.com/1.11/deploying-services/package-api/#/default/POST_package_describe :+1: